falcon/request_helpers.py

```diff
@@ -113,69 +118,59 @@ def fget(self):
     return property(fget)
 
 
-def make_etag(value, is_weak=False):
-    """Creates and returns a ETag object.
-
-    Args:
-        value (str): Unquated entity tag value
-        is_weak (bool): The weakness indicator
-
-    Returns:
-        A ``str``-like Etag instance with weakness indicator.
-
-    """
-    etag = ETag(value)
-    etag.is_weak = is_weak
-    return etag
+# NOTE(kgriffs): Going forward we should privatize helpers, as done here. We
+#   can always move this over to falcon.util if we decide it would be
+#   more generally useful to app developers.
+def _parse_etags(etag_str):
+    """Parse a string containing one or more HTTP entity-tags.
 
+    The string is assumed to be formatted as defined for a precondition
+    header, and may contain either a single ETag, or multiple comma-separated
+    ETags. The string may also contain a '*' character, in order to indicate
+    that any ETag should match the precondition.
 
-def parse_etags(etag_str):
-    """
-    Parse a string of ETags given in the If-Match or If-None-Match header as
-    defined by RFC 7232.
+    (See also: RFC 7232, Section 3)
 
     Args:
         etag_str (str): An ASCII header value to parse ETags from. ETag values
             within may be prefixed by ``W/`` to indicate that the weak comparison
             function should be used.
 
     Returns:
-        A list of unquoted ETags or ``['*']`` if all ETags should be matched.
+        list: A list of unquoted ETags or ``['*']`` if all ETags should be
+        matched. If the string to be parse is ``None``, empty, or contains only
+        whitespace, ``None`` will be returned instead.
 
```

Comment:
I added some additional cases to when ``None`` would be returned because it seemed to me the way a caller would deal with them would essential boil down to the same use case of "there wasn't an etag provided".